types/rails__activestorage/index.d.ts

```diff
@@ -6,22 +6,45 @@ export class DirectUpload {
     id: number;
     file: File;
     url: string;
+    delegate?: DirectUploadDelegate;
+    customHeaders: Record<string, string>;
 
-    constructor(file: File, url: string, delegate?: DirectUploadDelegate, customHeaders?: Record<string, string>);
+    constructor(
+        file: File,
+        url: string,
+        delegate?: DirectUploadDelegate,
+        customHeaders?: Record<string, string>,
+    );
 
     create(callback: (error: Error | null, blob?: Blob) => void): void;
+
+    // Dynamically assigned after create()
+    xhr?: XMLHttpRequest;
+    uploadRequest?: XMLHttpRequest;
+
+    // Optional hook assignable directly
+    directUploadWillCreateBlobWithXHR?: (xhr: XMLHttpRequest) => void;
 }
 
 export interface DirectUploadDelegate {
     directUploadWillCreateBlobWithXHR?: ((xhr: XMLHttpRequest) => void) | undefined;
-
     directUploadWillStoreFileWithXHR?: ((xhr: XMLHttpRequest) => void) | undefined;
 }
 
 export interface Blob {
+    signed_id: string;
+    filename: string;
+    content_type: string;
     byte_size: number;
     checksum: string;
-    content_type: string;
-    filename: string;
-    signed_id: string;
```

Comment:
Why should we touch this?

Reply:
You are right, I only meant to reorder them temporarily while I was working on this, didn't mean to check it in here.  Will revert, along with the empty line removal above.